.github/workflows/ci.yml

```diff
@@ -51,3 +52,38 @@ jobs:
       - run: cabal build
       - run: cabal test
       - run: ./script/haddock
+
+      - name: Create folder for artifacts
+        id: artifact
+        run: |
+          mkdir artifact
+          mkdir artifact/${{ matrix.os }}
+          mkdir artifact/${{ matrix.os }}/${{ matrix.ghc }}
+          echo ::set-output name=directory::'artifact/${{ matrix.os }}/${{ matrix.ghc }}'
+
+      - name: Generate package ready for Hackage distribution
+        run: cabal sdist --output-dir ${{ steps.artifact.outputs.directory }}
```

Comment:
@DavidMazarro what do you think about using `cabal sdist` default directory here? I think the default directory is `dist-newstyle/sdist`, that way you don't need to create the directory based on `{{ matrix.os }}` and `{{ matrix.ghc }}`